File: src/main/java/DaysJava/week1/Day6Anagram.java
package DaysJava.week1;

import java.util.Arrays;
import java.util.Scanner;

public class Day6Anagram {

    public static void main(String [] args){
        Day6Anagram anagram = new Day6Anagram();
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter the first string:");
        String s1 = scanner.nextLine();
        System.out.println("Enter the second string:");
        String s2 = scanner.nextLine();
        anagram.isAnagram(s1, s2);
    }

    private void isAnagram(String s1, String s2){
        if(s1.length() != s2.length()){
            System.out.println("These strings are not anagrams.");
        }

        char[] string1 = s1.toUpperCase().toCharArray();
        char[] string2 = s2.toUpperCase().toCharArray();

        Arrays.sort(string1);
        Arrays.sort(string2);

        if(Arrays.equals(string1, string2)){
            System.out.println(s1 + " and " + s2 + " are anagrams" );
        } else {
            System.out.println(s1 + " and " + s2 + " are not anagrams" );
        }
    }
}

File: src/main/kotlin/DaysKotlin/week1/Day6Anagram.kt
package DaysKotlin.week1

import java.util.*

fun main(){
    val reader = Scanner(System.`in`)
    println("Enter first String")
    val s1 = reader.nextLine()
    println("Enter second String")
    val s2 = reader.nextLine()
    anagram(s1, s2)
}

fun anagram(s1: String, s2: String){
    if(s1.length != s2.length) {
        println("These Strings are not anagrams")
    } else {
        var charString1 = s1.toUpperCase().toCharArray()
        var charString2 = s2.toUpperCase().toCharArray()

        charString1.sort()
        charString2.sort()
        if (charString1 contentEquals charString2) {
            println("$s1 and $s2 are anagrams")
        } else {
            println("$s1 and $s2 are not anagrams")
        }
    }
}